### PR TITLE
PLAT-601-2 Implemented BigDecimalFormatter

### PR DESCRIPTION
**Pull request by oliver-richers**

### CODE REVIEW

File: platform-common-core/src/test/java/com/softicar/platform/common/core/number/formatter/BigDecimalFormatterTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package com.softicar.platform.common.core.number.formatter;
+
+import com.softicar.platform.common.core.locale.ILocale;
+import com.softicar.platform.common.core.locale.Locale;
+import java.math.BigDecimal;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BigDecimalFormatterTest extends Assert {
```
Comment (alexander-schmidt) [inline platform-common-core/src/test/java/com/softicar/platform/common/core/number/formatter/BigDecimalFormatterTest.java:9]: This test does not, in any way, ensure that `BigDecimalFormatter` uses `CurrentLocale`. However, this is likely to be the most common use case.

We could even manipulate `BigDecimalFormatter`, replacing
`this.locale = CurrentLocale.get();`
with
`this.locale = null`
The thing would crash (unless `setLocale` gets called) - yet all tests will still be green.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-common-core/src/test/java/com/softicar/platform/common/core/number/formatter/BigDecimalFormatterTest.java (hunk 1 above)
Comment (oliver-richers) [reply platform-common-core/src/test/java/com/softicar/platform/common/core/number/formatter/BigDecimalFormatterTest.java:10]: did so; to made this testable, I introduced new `LocaleScope`

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleScope.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+public class LocaleScope implements AutoCloseable {
+
+	private final ILocale originalLocale;
+
+	/**
+	 * Sets {@link CurrentLocale} to the given {@link ILocale}.
+	 *
+	 * @param locale
+	 *            the language to set (never null)
```
Comment (alexander-schmidt) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleScope.java:16]:
```suggestion
	 *            the language to set (never <i>null</i>)
```

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleScope.java (hunk 2 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleScope.java]: fixed, also fixed `language` -> `ILocale`

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**